cmd/dfdaemon/app/root.go

```diff
@@ -41,45 +43,139 @@ var rootCmd = &cobra.Command{
 	Long:              "The dfdaemon is a proxy between container engine and registry used for pulling images.",
 	DisableAutoGenTag: true, // disable displaying auto generation tag in cli docs
 	RunE: func(cmd *cobra.Command, args []string) error {
-		initOption(opt)
+		if viper.GetBool("version") {
+			fmt.Println("dfdaemon version:", version.DFDaemonVersion)
+			return nil
+		}
+
+		if err := readConfigFile(viper.GetViper(), cmd); err != nil {
+			return errors.Wrap(err, "read config file")
+		}
 
-		s, err := dfdaemon.NewFromConfig(*g.Properties, *opt)
+		cfg, err := getConfigFromViper(viper.GetViper())
```

Comment:
It will cause to panic here. In fact, it's a common mistake for Golang Devs. FYI. http://devs.cloudimmunity.com/gotchas-and-common-mistakes-in-go-golang/index.html#nil_in_nil_in_vals

And you can use `!util.IsNil(err)` to avoid this problem.